### ISSUE TITLE
Bleadperl v5.19.3-26-gbbfdc87 breaks MUIR/modules/OOPS-0.2005.tar.gz

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13822]
Migrated from [rt.perl.org#121852](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=121852) (status was 'rejected')

Searchable as RT121852$

**Comment by p5pRT** [comment]
### From @andk

git bisect


---
commit bbfdc870734e1313430ade6e6bd6d8ee2b720413
Author&#8203;: Father Chrysostomos \<sprout@&#8203;cpan\.org>
Date&#8203;:   Tue Jul 16 22&#8203;:56&#8203;:44 2013 \-0700

&nbsp;    \[perl \#118691\] Allow defelem magic with neg indices

diagnostics


---
http&#8203;://www\.cpantesters\.org/cpan/report/bb22b228\-da05\-11e3\-a858\-50e272305bfd

ticket at rt\.cpan\.org


---
https://rt.cpan.org/Ticket/Display.html?id=95561

perl \-V


---
Summary of my perl5 \(revision 5 version 19 subversion 5\) configuration&#8203;:
&nbsp;  Commit id&#8203;: 1cd5f6726aaaf58dab89b5798e4c93b197e09b55
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=3\.13\-1\-amd64\, archname=x86\_64\-linux
&nbsp;    uname='linux k83 3\.13\-1\-amd64 \#1 smp debian 3\.13\.7\-1 \(2014\-03\-25\) x86\_64 gnulinux '
&nbsp;    config\_args='\-Dprefix=/home/sand/src/perl/repoperls/installed\-perls/perl/v5\.19\.5/165a \-Dmyhostname=k83 \-Dinstallusrbinperl=n \-Uversiononly \-Dusedevel \-des \-Ui\_db \-Uuseithreads \-Uuselongdouble \-DDEBUGGING=\-g'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=undef\, usemultiplicity=undef
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;    optimize='\-O2 \-g'\,
&nbsp;    cppflags='\-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.8\.2'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib/x86\_64\-linux\-gnu /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib /usr/lib
&nbsp;    libs=\-lnsl \-lgdbm \-ldb \-ldl \-lm \-lcrypt \-lutil \-lc \-lgdbm\_compat
&nbsp;    perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    libc=\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version='2\.18'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-O2 \-g \-L/usr/local/lib \-fstack\-protector'


Characteristics of this binary \(from libperl\)&#8203;: 
&nbsp;  Compile\-time options&#8203;: HAS\_TIMES PERLIO\_LAYERS PERL\_DONT\_CREATE\_GVSV
&nbsp;                        PERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME\_HARD PERL\_MALLOC\_WRAP
&nbsp;                        PERL\_NEW\_COPY\_ON\_WRITE PERL\_PRESERVE\_IVUV
&nbsp;                        PERL\_USE\_DEVEL USE\_64\_BIT\_ALL USE\_64\_BIT\_INT
&nbsp;                        USE\_LARGE\_FILES USE\_LOCALE USE\_LOCALE\_COLLATE
&nbsp;                        USE\_LOCALE\_CTYPE USE\_LOCALE\_NUMERIC USE\_PERLIO
&nbsp;                        USE\_PERL\_ATOF
&nbsp;  Built under linux
&nbsp;  Compiled at May  2 2014 07&#8203;:55&#8203;:37
&nbsp;  %ENV&#8203;:
&nbsp;    PERL5LIB=""
&nbsp;    PERL5OPT=""
&nbsp;    PERL5\_CPANPLUS\_IS\_RUNNING="7004"
&nbsp;    PERL5\_CPAN\_IS\_RUNNING="7004"
&nbsp;    PERL\_MM\_USE\_DEFAULT="1"
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.19\.5/165a/lib/site\_perl/5\.19\.5/x86\_64\-linux
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.19\.5/165a/lib/site\_perl/5\.19\.5
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.19\.5/165a/lib/5\.19\.5/x86\_64\-linux
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.19\.5/165a/lib/5\.19\.5
&nbsp;    \.
\-\- 
andreas

**Comment by p5pRT** [comment]
### From @iabyn

On Mon\, May 12\, 2014 at 02&#8203;:00&#8203;:09PM \-0700\, Andreas J\. Koenig via RT wrote&#8203;:
> git bisect
> \-\-\-\-\-\-\-\-\-\-
> commit bbfdc870734e1313430ade6e6bd6d8ee2b720413
> Author&#8203;: Father Chrysostomos \<sprout@&#8203;cpan\.org>
> Date&#8203;:   Tue Jul 16 22&#8203;:56&#8203;:44 2013 \-0700
> 
>     \[perl \#118691\] Allow defelem magic with neg indices
> 
> diagnostics
> \-\-\-\-\-\-\-\-\-\-\-
> http&#8203;://www\.cpantesters\.org/cpan/report/bb22b228\-da05\-11e3\-a858\-50e272305bfd
> 
> ticket at rt\.cpan\.org
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> https://rt.cpan.org/Ticket/Display.html?id=95561

This is a bug fix\.

The test in OOPS is testing for old buggy behaviour\. Basically its doing&#8203;:

&nbsp;    my @&#8203;a;
&nbsp;    $a\[7\] = 1;
&nbsp;    my $i = 4;
&nbsp;    foo\($a\[$i\]\);
&nbsp;    print "exists\\n" if exists $a\[$i\];

On 5\.18\.2 you get&#8203;: 'exists'
on 5\.19\.4 onwards you don't\.

On the other hand if you change it to '$i = 10'\, then
you don't get 'exists' on either\.

Basically that patch improved the autovivification avoidance for function
args from being just for elements above $\#a to being any non\-existent
element\.





\-\- 
Never do today what you can put off till tomorrow\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @andk

"Dave Mitchell via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:

> This is a bug fix\.

Thanks a lot for the assessment\!

\-\- 
andreas

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'rejected'
